src/components/Calculator.js

```diff
@@ -1,64 +1,60 @@
-          <Button btnClass="gray-btn" input="1" handleClickButton={this.handleClickButton} />
-          <Button btnClass="gray-btn" input="2" handleClickButton={this.handleClickButton} />
-          <Button btnClass="gray-btn" input="3" handleClickButton={this.handleClickButton} />
-          <Button btnClass="orange-btn" input="+" handleClickButton={this.handleClickButton} />
-        </div>
-        <div className="row">
-          <Button btnClass="gray-btn col-md-2" input="0" handleClickButton={this.handleClickButton} />
-          <Button btnClass="gray-btn" input="." handleClickButton={this.handleClickButton} />
-          <Button btnClass="orange-btn" input="=" handleClickButton={this.handleClickButton} />
-        </div>
-      </div>
-    );
+  const { total, next } = state;
+  let result = 0;
+  if (next !== null) {
+    result = next;
+  } else if (total !== null) {
+    result = total;
   }
-}
+  return (
+    <div className="calc-container">
+      <div className="row">
+        <span className="result text-end col-md-4 py-3">{result}</span>
+      </div>
+      <div className="row">
+        <Button btnClass="gray-btn" input="AC" handleClickButton={handleClickButton} />
+        <Button btnClass="gray-btn" input="+/-" handleClickButton={handleClickButton} />
+        <Button btnClass="gray-btn" input="%" handleClickButton={handleClickButton} />
+        <Button btnClass="orange-btn" input="÷" handleClickButton={handleClickButton} />
+      </div>
+      <div className="row">
+        <Button btnClass="gray-btn" input="7" handleClickButton={handleClickButton} />
+        <Button btnClass="gray-btn" input="8" handleClickButton={handleClickButton} />
+        <Button btnClass="gray-btn" input="9" handleClickButton={handleClickButton} />
+        <Button btnClass="orange-btn" input="x" handleClickButton={handleClickButton} />
+      </div>
+      <div className="row">
+        <Button btnClass="gray-btn" input="4" handleClickButton={handleClickButton} />
+        <Button btnClass="gray-btn" input="5" handleClickButton={handleClickButton} />
+        <Button btnClass="gray-btn" input="6" handleClickButton={handleClickButton} />
+        <Button btnClass="orange-btn" input="-" handleClickButton={handleClickButton} />
+      </div>
+      <div className="row">
+        <Button btnClass="gray-btn" input="1" handleClickButton={handleClickButton} />
+        <Button btnClass="gray-btn" input="2" handleClickButton={handleClickButton} />
+        <Button btnClass="gray-btn" input="3" handleClickButton={handleClickButton} />
+        <Button btnClass="orange-btn" input="+" handleClickButton={handleClickButton} />
+      </div>
+      <div className="row">
+        <Button btnClass="gray-btn col-md-2" input="0" handleClickButton={handleClickButton} />
+        <Button btnClass="gray-btn" input="." handleClickButton={handleClickButton} />
+        <Button btnClass="orange-btn" input="=" handleClickButton={handleClickButton} />
```

Comment:
- [ ] [OPTIONAL] There is a way to avoid all these `onClick` listeners and use only one, this technique is called `bubbling up` 😄 
    - I will leave here a link to a [youtube tutorial](https://www.youtube.com/watch?v=SqQZ8SttQsI),  this could be helpful in this project and in future projects 🚀

Reply:
- [ ] [OPTIONAL] Instead of creating each button manually you could create an array with objects, that hold that button data, and then display the buttons with `map` something like this:
```javascript
    {buttonsArr.map((button) => (
       <Button content={button. content} className={button.className} onClick={button.event} 
    ))

```